R/derive_merged.R

```diff
@@ -178,77 +188,101 @@
 #' derive_vars_merged(
 #'   select(admiral_dm, STUDYID, USUBJID),
 #'   dataset_add = ex_ext,
+#'   filter_add = !is.na(EXSTDTM),
 #'   by_vars = exprs(STUDYID, USUBJID),
 #'   new_vars = exprs(TRTSDTM = EXSTDTM, TRTSDTF = EXSTDTF, TRTSTMF = EXSTTMF),
 #'   order = exprs(EXSTDTM),
 #'   mode = "first"
 #' )
 #'
-#' # Derive treatment start datetime (TRTSDTM)
-#' data(admiral_ex)
-#'
-#' ## Impute exposure start date to first date/time
-#' ex_ext <- derive_vars_dtm(
-#'   admiral_ex,
-#'   dtc = EXSTDTC,
-#'   new_vars_prefix = "EXST",
-#'   highest_imputation = "M",
-#' )
-#'
-#' ## Add first exposure datetime and imputation flags to adsl
+#' # Derive treatment end datetime (TRTEDTM)
+#' # The conversion from DTC to DTM is done within new_vars
+#' # The new variable can be used in filter_add and order
 #' derive_vars_merged(
 #'   select(admiral_dm, STUDYID, USUBJID),
 #'   dataset_add = ex_ext,
-#'   filter_add = !is.na(EXSTDTM),
+#'   filter_add = !is.na(TRTEDTM),
 #'   by_vars = exprs(STUDYID, USUBJID),
-#'   new_vars = exprs(TRTSDTM = EXSTDTM, TRTSDTF = EXSTDTF, TRTSTMF = EXSTTMF),
-#'   order = exprs(EXSTDTM),
-#'   mode = "first"
+#'   new_vars = exprs(TRTEDTM = convert_dtc_to_dtm(EXENDTC)),
+#'   order = exprs(TRTEDTM),
+#'   mode = "last"
 #' )
 #'
-#' # Derive treatment end datetime (TRTEDTM)
-#' ## Impute exposure end datetime to last time, no date imputation
-#' ex_ext <- derive_vars_dtm(
-#'   admiral_ex,
-#'   dtc = EXENDTC,
-#'   new_vars_prefix = "EXEN",
-#'   time_imputation = "last",
-#' )
+#' # Modify merged values and set value for non matching observations
+#' adsl <- tribble(
+#'   ~USUBJID, ~SEX, ~COUNTRY,
+#'   "ST42-1", "F",  "AUT",
+#'   "ST42-2", "M",  "MWI",
+#'   "ST42-3", "M",  "NOR",
+#'   "ST42-4", "F",  "UGA"
+#' ) %>% mutate(STUDYID = "ST42")
```

Comment:
Is StudyID needed?

Reply:
No, I have removed `STUDYID` to keep it simple.